### ISSUE TITLE
Errors when training with ResNet-50 model

### ISSUE THREAD

**Issue by michaelhuang74** [jcjohnson/fast-neural-style#111]
Firstly, I tried to convert the prototxt and the Caffe model of ResNet-50 downloaded from https://github.com/KaimingHe/deep-residual-networks to torch model using the following command:

th convert_model.lua  -input_prototxt resnet50/ResNet-50-deploy.prototxt -input_caffemodel resnet50/ResNet-50-model.caffemodel -output_t7 resnet50.t7

However, it seemed that loadcaffe does not support some modules in the caffemodel of ResNet-50. The conversion was not successful.

[libprotobuf ERROR google/protobuf/text_format.cc:245] Error parsing text-format caffe.NetParameter: 26:26: Message type "caffe.LayerParameter" has no field named "batch_norm_param".
Successfully loaded ../resnet50/ResNet-50-model.caffemodel
warning: module 'bn_conv1 [type BatchNorm]' not found
warning: module 'scale_conv1 [type Scale]' not found
warning: module 'pool1_pool1_0_split [type Split]' not found
warning: module 'bn2a_branch1 [type BatchNorm]' not found
..........
warning: module 'scale5c_branch2c [type Scale]' not found
warning: module 'res5c [type Eltwise]' not found
conv1: 64 3 7 7
res2a_branch1: 256 64 1 1
Segmentation fault (core dumped)

Then I downloaded the torch model of ResNet-50 by facebook from https://github.com/facebook/fb.resnet.torch. I obtained the layer information from the ResNet-50-deploy.prototxt (which is from https://github.com/KaimingHe/deep-residual-networks). The ReLU layers are listed in 
[resnet-50-relu.txt](https://github.com/jcjohnson/fast-neural-style/files/870556/resnet-50-relu.txt)

Followings are the first few ReLU layers:
4 conv1_relu
12 res2a_branch2a_relu
16 res2a_branch2b_relu
21 res2a_relu
25 res2b_branch2a_relu
29 res2b_branch2b_relu
34 res2b_relu

When I tried to train a style image using the ResNet-50.t7 from facebook, say, using res2a_relu for both style layer and content layer, I encountered an error.

The command:
th train.lua -h5_file file.h5 -style_image style/face_bottle_o.png -style_image_size 512 -style_weights 15 -style_layers 21 -content_layers 21 -loss_network models/resnet-50.t7 -preprocessing resnet -checkpoint_name checkpoint/face_bottle_o.s512.sw15.resnet50.gram.4x10000 -num_iterations 10000 

The error message:
/home/mqhuang/torch/install/bin/luajit: /home/mqhuang/torch/install/share/lua/5.1/nn/Container.lua:63: attempt to index local 'module' (a nil value)
stack traceback:
        /home/mqhuang/torch/install/share/lua/5.1/nn/Container.lua:63: in function 'rethrowErrors'
        /home/mqhuang/torch/install/share/lua/5.1/nn/Sequential.lua:58: in function 'updateGradInput'
        ./fast_neural_style/PerceptualCriterion.lua:159: in function 'backward'
        train.lua:199: in function 'opfunc'
        /home/mqhuang/torch/install/share/lua/5.1/optim/adam.lua:37: in function 'adam'
        train.lua:239: in function 'main'
        train.lua:327: in main chunk
        [C]: in function 'dofile'
        ...uang/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
        [C]: at 0x00406670

Any idea how to resolve this issue? Thanks.
